traffic_control/views.py

```diff
@@ -103,3 +110,29 @@ class RoadMarkingPlanViewSet(TrafficControlViewSet):
 class RoadMarkingRealViewSet(TrafficControlViewSet):
     serializer_class = RoadMarkingRealSerializer
     queryset = RoadMarkingReal.objects.all()
+
+
+class TrafficSignCodeViewSet(ModelViewSet):
+    filter_backends = [DjangoFilterBackend, OrderingFilter]
+    ordering_fields = "__all__"
+    ordering = ["code"]
+    permission_classes = [IsAdminUser]
+    serializer_class = TrafficSignCodeSerializer
+    queryset = TrafficSignCode.objects.all()
+
+    def create(self, request, *args, **kwargs):
```

Comment:
I'm not totally convinced that this is actually even needed. You could just have the "validation" in the model's clean method - if it fails then it should return 400 as well. In this case in the code has to be unique, then just have unique = true in the model or something like that and you don't even have to do the custom clean in the model.